### PR TITLE
Add authentication options for import

### PR DESCRIPTION
**Pull request by jjeising**
This adds options for HTTP basic auth and a user definable `Authentication` header for import.

Authentication options are currently reused when repeating an import, but are not changeable or viewable.

### CODE REVIEW

File: src/Application/Controller/Import.php [hunk 1]
```diff
@@ -106,21 +107,22 @@ public function repeat(array $arguments) {
...
+					$xml = $this->_getXMLAndParse($import->toArray());
 					
 					if ($xml === false) {
 						return $this->Response;
 					}
 				}
 				
-				$import = Import::createOrThrow([
-					'project_id' => $this->project['id'],
+				$import = $this->import->duplicate(true);
```
Comment (pegro) [inline src/Application/Controller/Import.php]: Is the call to duplicate in line 107 not enough?

**Comment by zuntrax** [comment]
When you create a new import with "No Authentication" selected, the following error occurs:

```
Server error
InvalidDataException

ERROR: invalid input value for enum enum_import_auth_type: ""
```

**Comment by jjeising** [comment]
> When you create a new import with "No Authentication" selected, the following error occurs:

Thanks for testing and reporting!

This issue occurs because we currently don't handle this field as `Database::TYPE_ENUM` because `information_schema.columns.data_type` is `USER-DEFINED`:

```
    array (size=6)
      'column_name' => string 'auth_type' (length=9)
      'data_type' => string 'USER-DEFINED' (length=12)
      'is_nullable' => string 'YES' (length=3)
      'column_default' => null
      'character_maximum_length' => null
      'numeric_precision' => null
```

We should improve handling here, set the type and add available enum options, like we do for MySQL.
@pegro Can you give some PostgreSQL insight here?

The other change that would fix this the option to nullify blank string values (string is the fallback type here) in newer framework versions. But this is an additional Model option, so it is less desirable.

**Comment by pegro** [comment]
You could check the value in information_schema.columns.udt_name for user-defined types, as mentioned in the documentation (https://www.postgresql.org/docs/current/static/infoschema-columns.html). Does it contain something you could use?
If you don't want to add PHP code for that you could extend the query and use `
CASE WHEN data_type = 'USER-DEFINED' THEN udt_name ELSE data_type END AS data_type`

**Comment by jjeising** [comment]
> Does it contain something you could use?

`udt_name` gives `enum_import_auth_type` – this is not very useful, especially since the name does not have to contain *enum*. [As far, as I understand](https://stackoverflow.com/a/6245537) need need to adapt the query for PostgreSQL. This isn't a problem, as PostgreSQL and MySQL already have separate queries and handling (we use `SHOW COLUMNS…` for MySQL). [`enum_range()`](https://stackoverflow.com/questions/1616123/sql-query-to-get-all-values-a-enum-can-have/1616161) looks nice, but I'm not sure if it supports string field names. In the end we probably need to query `pg_` tables, ideally producing a more or less similar result to MySQL (type *enum* and a field with comma separated values (or a more useful data format)).

**Comment by a-tze** [comment]
Improving the framework is a good idea.. but what about changing the auth_type column to
`not null default 'none'`
and add 'none' as enum value?

**Comment by jjeising** [comment]
> but what about changing the auth_type column to `not null default 'none'` and add 'none' as enum value?

I'm not quite sure what I think about *none*. We would also have to set this value in the dropdown, otherwise we would still send an empty string.

**Comment by jjeising** [comment]
@pegro I've created [an issue with a new query](https://github.com/selah/selah/issues/1), if you would like to have a look.

**Comment by jjeising** [comment]
Fixed issue with enums, would merge now. Any objections  @pegro, @a-tze ?

**Pull request merged**